torch/csrc/jit/pickler.cpp

```diff
@@ -403,8 +404,41 @@ void Pickler::pushLiteralTensor(const IValue& ivalue) {
   push<PickleOpCode>(PickleOpCode::TUPLE);
 
   if (quantized) {
-    pushDouble(tensor.q_scale());
-    pushInt(tensor.q_zero_point());
+    push<PickleOpCode>(PickleOpCode::MARK);
+    pushGlobal("torch", toString(tensor.qscheme()));
+    // tuple of (qscheme, scale, zp) or (qscheme, scales, zps, axis)
```

Comment:
Why are there two different schemas here? Is it possible to always do it as `(qscheme, scales, zps, axis)`? Specializing is going to make the (de)serialization logic much more complicated

Reply:
It depends on quantizer (which can be per-tensor or per-channel) - the per-tensor one doesn't have axis. I can make axis to be None in the other case, but not sure it makes it cleaner. Specialization still has to be there as the scale vs scales is int vs tuple of int